Added pre-requisite for macOS

abhi3700 opened the pull request:
Greetings!

This PR introduces instructions tailored for macOS, ensuring all prerequisites are met before proceeding to the build step using:
```sh
$ cargo build
```

The instructions have been organized into separate sections for **Ubuntu** and **macOS**.

### Code contributor checklist:
* [x] I have read, understood and followed [contributing guide](https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md)

abhi3700 committed:
added macOS M1 example for ensuring the dependencies before 'cargo build'

nazar-pc commented on docs/development.md:
````diff
@@ -1,16 +1,48 @@
+
+   ```sh
+   $ llvm-config --version
+   ```
````
This is not specific for M1 in any way, so let's remove M1 mention. Also no need for extra indentation of code snippets and we did use `bash` rather than `sh` in all other examples and without `$ `, so it would be nice to keep it consistent.

nazar-pc reviewed (commented):
Some nits, makes sense otherwise.

abhi3700 committed:
minor changes

- codeblock indentation removed
- sh to bash
- made steps generic for all macOS irrespective of hardware
- removed $ sign
